Reverification iOS support and refactor

wedaly opened the pull request:
# Overview

Adds iOS support to the reverification flow, cleaning up a lot of cruft along the way.  

* Delete reverification templates
* Delete photocapture.js
* Delete unused "name change" end-points
* Rebuild the reverification views using Backbone sub-views
* Stop passing template names to the JavaScript code
* Avoid hard-coding the parent view ID in the webcam view (for getting the capture click sound URL)

# Reviewers

@clintonb please review the code changes

@frrrances when you have a moment, I'd like to check in about what styles I can re-use to make the above screenshots look decent.

FYI: @cahrens I went ahead and deleted the "change name" end-points we talked about, since `photocapture.js` was the only thing left in the system using them.

FYI: @srpearce This is the ticket I mentioned yesterday.  The screenshots above show placeholder text that I could use some guidance on.

cahrens commented:
Thanks for doing the deletion, @wedaly! I had previously looked at Splunk to see if anyone was calling them, and only saw Jenkins test usage over the last 30 days.

wedaly commented:
@cahrens That sounds about right.  It's next-to-impossible to get into the reverification flow anyway, and the "change name" modal is pretty hidden once you get there :)

wedaly commented:
jenkins test this please

cahrens commented:
@wedaly is this PR still active?

wedaly commented:
@cahrens Yes, I'm updating some of the styling now, then waiting on reviews from @clintonb and @srpearce.

wedaly commented:
@srpearce @frrrances I've updated the styling and eliminated one of the states.  This is what it looks like now:

* Successfully submitted photos for reverification
![image](https://cloud.githubusercontent.com/assets/2948394/8215747/3740b86a-1500-11e5-8062-da3a87cb2c89.png)

* Already have an approved or pending verification:
![image](https://cloud.githubusercontent.com/assets/2948394/8215751/3c6517aa-1500-11e5-9af9-e933015f895f.png)

* No initial verification (got to this page by directly visiting the URL):
![image](https://cloud.githubusercontent.com/assets/2948394/8215753/3e57f014-1500-11e5-94a4-b40c7cea192c.png)

@frrrances I also found and deleted a bunch of sass from the midcourse reverification flows that was no longer being used.  I'm in the process of testing the existing verification flows to make sure this hasn't accidentally broken anything (although I searched for the class names and didn't find anything).

frrrances commented:
Sass cleanup looks good. :+1:

srpearce commented:
Working on the messages now...

srpearce commented on lms/templates/verify_student/reverify_success_step.underscore:
```diff
@@ -0,0 +1,11 @@
+    <h1 class="title"><%- gettext( "Your Credentials Have Been Updated" ) %></h3>
+
+    <div class="instruction">
+        <p><%- gettext( "We've captured your re-submitted information and will review it to verify your identity shortly. You should receive an update to your veriication status within 1-2 days. In the meantime, you still have access to all of your course content." ) %></p>
```
We have received your information and are verifying your identity. You will see a message on your dashboard when the verification process is complete (usually within 1-2 days). In the meantime, you can still access all available course content.

srpearce commented on lms/templates/verify_student/reverify_success_step.underscore:
```diff
@@ -0,0 +1,11 @@
+<div id="wrapper-reverify-success" tab-index="0" class="wrapper-view reverify-success-step">
+    <h1 class="title"><%- gettext( "Your Credentials Have Been Updated" ) %></h3>
```
Identity Verification In Progress

srpearce commented on lms/templates/verify_student/reverify_not_allowed.html:
```diff
@@ -0,0 +1,28 @@
+            % if status in ["pending", "approved"]:
+                ${_("You have already submitted a verification attempt, so you don't need to reverify.")}
+            % else:
+                ${_("You cannot reverify at this time.")}
```
You cannot verify your identity at this time.

@wedaly Can you tell me why a user wouldn't be able to reverify? That might change the message we want to show.

wedaly replied:
This is a catch-all if the user somehow visits this URL directly without having gone through the verification flow.  Very, very few users should ever see this.

srpearce commented on lms/templates/verify_student/reverify_not_allowed.html:
```diff
@@ -0,0 +1,28 @@
+    <div class="instructions">
+        <p>
+            % if status in ["pending", "approved"]:
+                ${_("You have already submitted a verification attempt, so you don't need to reverify.")}
```
You have already submitted your verification information. You will see a message on your dashboard when your identity has been verified (usually within 1-2 days).

srpearce commented on lms/templates/verify_student/reverify_not_allowed.html:
```diff
@@ -0,0 +1,28 @@
+
+<%block name="content">
+<div class="reverify-blocked">
+    <h1 class="title">${_("Reverification")}</h1>
```
Identity Verification

srpearce commented on lms/templates/verify_student/reverify_not_allowed.html:
```diff
@@ -0,0 +1,28 @@
+%>
+
+<%inherit file="../main.html" />
+<%block name="pagetitle">${_("Re-Verification")}</%block>
```
Identity Verification

srpearce commented:
Message text review complete.

wedaly commented:
Thanks, @srpearce !  Updated.

wedaly commented:
@clintonb This is ready for a code review when you have a moment.  I'd like to get this in before I leave, but if not I can pick it up next week.

wedaly commented:
test this please

clintonb commented on lms/djangoapps/verify_student/tests/test_views.py:
```diff
@@ -42,8 +42,8 @@
     checkout_with_ecommerce_service,
-    render_to_response, PayAndVerifyView, EVENT_NAME_USER_ENTERED_INCOURSE_REVERIFY_VIEW,
-    EVENT_NAME_USER_SUBMITTED_INCOURSE_REVERIFY, _send_email, _compose_message_reverification_email
+    render_to_response, PayAndVerifyView,
```
Minor: these three lines can be merged into 1 or 2.

clintonb commented on lms/djangoapps/verify_student/views.py:
```diff
@@ -1044,85 +1008,52 @@ def results_callback(request):
-    """
-    # TODO need someone to review the copy for this template
-    return render_to_response("verify_student/reverification_window_expired.html")
+            return render_to_response("verify_student/reverify.html", context)
```
[TemplateView](https://docs.djangoproject.com/en/1.8/ref/class-based-views/base/#templateview) could help clean some of this up, but introduces a new problem since you have to switch templates based on the model lookup. I guess you could look that up once and store it as a `cachedproperty`...

```python
class ReverifyView(TemplateView):
  @method_decorator(login_required)
  def dispatch(...)
    return super(...).dispatch(...)

  def get_template_names(self):
    if self.status in ["must_reverify", "expired"]:
      return ['verify_student/reverify.html']
    return ['verify_student/reverify_not_allowed.html']

  # Not sure if cached_property in Django 1.4
  @cached_property
  def status(self):
    status, _ = SoftwareSecurePhotoVerification.user_status(request.user)
    return status

  def get_context_data(...):
    context = super(ReverifyView, self).get_context_data(...)
    context.update({
      'status': self.status,
      'user_full_name': request.user.profile.name,
      'platform_name': settings.PLATFORM_NAME,
      'capture_sound': staticfiles_storage.url("audio/camera_capture.wav"),
    });
    return context
```

clintonb commented:
:+1:

wedaly commented:
Will merge once tests pass.

wedaly commented:
Test failure is unrelated (JavaScript test could not retrieve results).

benpatterson commented:
@wedaly unfortunately the JS error on this PR is consistently taking place on master now. Do you have time to look at it? We can also revert.

[I also acknowledge that condition can be a flaky condition...unfo it appears to be consistent at this point.]
